src/display.cpp

```diff
@@ -2211,6 +2236,21 @@ double display::turbo_speed() const
 		return 1.0;
 }
 
+void display::set_prevent_draw(bool pd)
+{
+	prevent_draw_ = pd;
+	if (!pd) {
+		// ensure buttons are visible
+		unhide_buttons();
+	}
```

Comment:
Shouldn't the else branch then call`hide_buttons()`?

Reply:
At that point the buttons don't exist yet. I considered making it hide, but it might actually cause strange behaviour with disappearing buttons if someone uses it mid-game for some reason.

Are there any scenarios that trigger the story-viewer mid-scenario? I'm not sure how it would behave in that case.

Reply:
I remembered there was an event in DiD that did. In this case they don't need to hide as the story screen correctly covers the buttons, it's only when the buttons are initially created that they jump on top of everything else.